### PR TITLE
James 1914

### PR DESCRIPTION
**Pull request by chibenwa**
Yoloooo ! 

https://github.com/linagora/james-project/pull/582

Only last commit.

### CODE REVIEW

**Comment by chibenwa** [comment]
Now compiles.

But what the hell is that ?

```
[Building Java 8 james-1914-16d24f69-f542-47a8-996f-3ff7c791d7e6] Results :
[Building Java 8 james-1914-16d24f69-f542-47a8-996f-3ff7c791d7e6] 
[Building Java 8 james-1914-16d24f69-f542-47a8-996f-3ff7c791d7e6] Tests in error: 
[Building Java 8 james-1914-16d24f69-f542-47a8-996f-3ff7c791d7e6]   TODO: implement me
[Building Java 8 james-1914-16d24f69-f542-47a8-996f-3ff7c791d7e6]   TODO: implement me
[Building Java 8 james-1914-16d24f69-f542-47a8-996f-3ff7c791d7e6] 
[Building Java 8 james-1914-16d24f69-f542-47a8-996f-3ff7c791d7e6] Tests run: 701, Failures: 0, Errors: 2, Skipped: 1
```

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by strokyl** [comment]
test this please

File: mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/InMemoryMessageIdManager.java [hunk 1]
```diff
@@ -112,8 +113,11 @@ public void setInMailboxes(MessageId messageId, List<MailboxId> mailboxIds, Mail
             for (MailboxId mailboxId: mailboxesToAdd) {
                 MessageRange messageRange = referenceMessage.getUid().toRange();
                 mailboxManager.copyMessages(messageRange, referenceMessage.getMailboxId(), mailboxId, mailboxSession);
+                Flags flags = new Flags(referenceMessage.getFlags());
+                System.out.println("###########" + flags.contains(Flag.RECENT)); //weirdly it print false here but in the in memory integration test still work
```
Comment (strokyl) [inline mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/InMemoryMessageIdManager.java]: So here when doing a test in memory and when doing a move there is no recent flag but the integration test pass

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java [hunk 2]
```diff
@@ -223,6 +224,7 @@ private void addMessageToMailboxes(MessageIdMapper messageIdMapper, MailboxMessa
         for (MailboxId mailboxId : mailboxIds) {
             SimpleMailboxMessage copy = SimpleMailboxMessage.copy(mailboxId, mailboxMessage);
             MessageMetaData metaData = save(mailboxSession, messageIdMapper, copy);
+            metaData.getFlags().add(Flag.RECENT);
```
Comment (strokyl) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: This fix the integration test about reading in IMAP a JMAP move

File: protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java [hunk 3]
```diff
@@ -406,9 +406,13 @@ public synchronized void event(Event event) {
                 final MessageEvent messageEvent = (MessageEvent) event;
                 if (messageEvent instanceof Added) {
                     sizeChanged = true;
-                    final List<MessageUid> uids = ((Added) event).getUids();
+                    Added added = (Added) event;
+                    final List<MessageUid> uids = added.getUids();
                     for (MessageUid uid : uids) {
                         add(uid);
+                        if (added.getMetaData(uid).getFlags().contains(Flag.RECENT)) {
+                            addRecent(uid);
```
Comment (strokyl) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java]: This fix the integration test about reading in IMAP a IMAP copy

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java (hunk 2 above)
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: I don't understand why you don't save it as recent (why this line is not before save call) ?

Comment (strokyl) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java]: Because I am stupid. Done

**Comment by aduprat** [comment]
#597

**Comment by aduprat** [comment]
Merged: #597

**Pull request closed without merging**